crates/connector_proxy/src/apis.rs

```diff
@@ -38,30 +38,68 @@ impl FlowOperation for FlowMaterializeOperation {}
 // InterceptorStream defines the type of input and output streams handled by interceptors.
 pub type InterceptorStream = Pin<Box<dyn Stream<Item = std::io::Result<Bytes>> + Send + Sync>>;
 
-// The generic param "T" below is bounded by FlowOperation.
-// A converter is a function that contains the specific stream-handling logic of an interceptor.
-type ConverterFn<T> = Box<dyn Fn(&T, InterceptorStream) -> Result<InterceptorStream, Error>>;
-// An intercept is characterized by a pair of converters, corresponding to the handling logic of request and response streams, respectively.
-pub type RequestResponseConverterPair<T> = (ConverterFn<T>, ConverterFn<T>);
 pub trait Interceptor<T: FlowOperation> {
-    fn get_converters() -> RequestResponseConverterPair<T> {
-        (
-            Box::new(|_op, stream| Ok(stream)),
-            Box::new(|_op, stream| Ok(stream)),
-        )
+    fn convert_command_args(&mut self, _op: &T, args: Vec<String>) -> Result<Vec<String>, Error> {
+        Ok(args)
+    }
+
+    fn convert_request(
+        &mut self,
+        // The pid of the connector process, which is in stopped state at first, waiting for SIGCONT signals from interceptors.
+        pid: Option<u32>,
```

Comment:
pass in `pid` in the `convert_request`, so that the connector process could be started after the interceptors.

The reason is that that arguments used to start the connectors process are constructed by the interceptor, so it is easier if we avoid the interceptor to depend on the connector process to be started.